proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java

```diff
@@ -451,15 +454,18 @@ public Response postPushData(UUID agentId, UUID workUnitId, Long currentMillis,
   }
 
   private void addTaskToSmallestQueue(ResubmissionTask taskToRetry) {
-    TaskQueue<ResubmissionTask> queue = getSmallestQueue();
+    ResubmissionTaskQueue queue = getSmallestQueue();
     if (queue != null) {
-      synchronized (queue) {
+      try {
```

Comment:
also, it's normally lock then try ... finally {unlock} (not that it matters i believe in practice but let's say it OOM right after you enter the try block...)